lib/new_relic/agent/instrumentation/logger/instrumentation.rb

```diff
@@ -39,6 +39,7 @@ def format_message_with_tracing(severity, datetime, progname, msg)
             mark_skip_instrumenting
 
             unless ::NewRelic::Agent.agent.nil?
+              formatted_message = LocalLogDecorator.decorate(formatted_message)
```

Comment:
I'm a little hesitant about putting the instrumentation here. I keep going back and forth with the `LogEventAggregator`. Though it's a rare case, if someone has forwarding and decoration enabled, and the message is longer than 32 kb, the decoration would be truncated. If it's in the LogEventAggregator, a lot of extra code is run to add decoration that may not be necessary. 

One idea could be to add some complexity to the truncation, where it finds the size of the log decoration, truncates the message for decoration_size - max bytes and re-applies the decoration with the truncated message. This seems like it might be too much overhead for each logging line.

Thoughts?

Reply:
So my perspective here -- I'd say that we should be doing it in the instrumentation vs the aggregator... but I think we should do it _after_ we've done record.

If we're sending from the agent, we don't need the linking in the message as it'll be tacked on with the collector post. We also don't want that our truncating of what we're sending up to NR ends up truncating their local output, as they may care about that being complete.